gobblin-metastore/src/test/java/org/apache/gobblin/metastore/testing/TestMetastoreDatabaseServer.java

```diff
@@ -83,27 +81,23 @@ class TestMetastoreDatabaseServer implements Closeable {
     this.embeddedMysqlEnabled = realConfig.getBoolean(EMBEDDED_MYSQL_ENABLED_KEY);
     this.dbUserName = realConfig.getString(DBUSER_NAME_KEY);
     this.dbUserPassword = realConfig.getString(DBUSER_PASSWORD_KEY);
-    this.dbHost = this.embeddedMysqlEnabled ? "localhost" : realConfig.getString(DBHOST_KEY);
-    this.dbPort = this.embeddedMysqlEnabled ? new PortUtils.ServerSocketPortLocator().random() : realConfig.getInt(DBPORT_KEY);
-
-    this.log.info("Starting with config: embeddedMysqlEnabled={} dbUserName={} dbHost={} dbPort={}",
-                  this.embeddedMysqlEnabled,
-                  this.dbUserName,
-                  this.dbHost,
-                  this.dbPort);
-
-    config = MysqldConfig.aMysqldConfig(Version.v8_latest)
-        .withPort(this.dbPort)
-        .withUser(this.dbUserName, this.dbUserPassword)
-        .withServerVariable("explicit_defaults_for_timestamp", "off")
-        // default `max_connections` is apparently 151 - see: https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_max_connections
-        .withServerVariable("max_connections", "501")
-        .build();
     if (this.embeddedMysqlEnabled) {
-      testingMySqlServer = EmbeddedMysql.anEmbeddedMysql(config).start();
-    }
-    else {
-      testingMySqlServer = null;
+      mySQLContainer = new MySQLContainer<>("mysql:" + MYSQL_VERSION)
+          .withUsername(this.dbUserName)
+          .withPassword(this.dbUserPassword)
+          .withEnv(MYSQL_ROOT_PASSWORD, "");
+
+      mySQLContainer.start();
+      this.dbHost = mySQLContainer.getHost();
+      this.dbPort = mySQLContainer.getFirstMappedPort();
+
+      this.log.info("Started with config: embeddedMysqlEnabled={} dbUserName={} dbHost={} dbPort={}",
+          this.embeddedMysqlEnabled, this.dbUserName, this.dbHost, this.dbPort);
+    } else {
+      mySQLContainer = null;
+      this.dbHost = this.embeddedMysqlEnabled ? "localhost" : realConfig.getString(DBHOST_KEY);
```

Comment:
Conditional check not needed since here the condition will always be false.

Reply:
addressed